src/main/java/com/epam/page/object/generator/validators/AbstractValidator.java

```diff
@@ -1,129 +1,53 @@
 package com.epam.page.object.generator.validators;
 
-import com.epam.page.object.generator.model.SearchRule;
-import com.epam.page.object.generator.utils.SearchRuleType;
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Set;
+import com.epam.page.object.generator.models.searchrule.CommonSearchRule;
+import com.epam.page.object.generator.models.searchrule.ComplexInnerSearchRule;
+import com.epam.page.object.generator.models.searchrule.ComplexSearchRule;
+import com.epam.page.object.generator.models.searchrule.FormInnerSearchRule;
+import com.epam.page.object.generator.models.searchrule.FormSearchRule;
+import com.epam.page.object.generator.models.webgroup.CommonWebElementGroup;
+import com.epam.page.object.generator.models.webgroup.ComplexWebElementGroup;
+import com.epam.page.object.generator.models.webgroup.FormWebElementGroup;
 
-/**
- * AbstractValidator is the abstract base class for crating a new Validator, which can be used into
- * ValidatorsStarter. <p> For creating a new Validator you need: <ol> <li>Create a new class which
- * extends from {@link AbstractValidator} and override the main method {@link
- * AbstractValidator#isValid(SearchRule, ValidationContext)}.</li> <li>Create default constructor
- * with {@link AbstractValidator#priority}.</li> <li>Override {@link
- * AbstractValidator#getExceptionMessage(SearchRule searchRule, ValidationContext
- * validationContext)}.</li> </ol>
- */
-public abstract class AbstractValidator implements Validator {
+public abstract class AbstractValidator implements ValidatorVisitor {
```

Comment:
It is pain to see documentation level decrease. Provide javadoc for at least key classes of affected files.

Reply:
Done